### PR TITLE
Site Picker: Change wording of /page to match /post

### PR DESCRIPTION
**Pull request by codebykat**
#### Changes proposed in this Pull Request

I noticed that if you go to https://wordpress.com/post, the prompt says "Select a site to start writing":

screen shot 2019-01-01 at 5 15 19 pm

However, on https://wordpress.com/page, it falls back to the default "Please select a site to open Page":
    
screen shot 2019-01-01 at 5 15 31 pm

This PR uses the same, nicer wording on the `/page` page :)

Technical note: The code works, but I did notice that the Post wording seems to be coming from the [Press This part of the post-editor controller](https://github.com/Automattic/wp-calypso/blob/39e3fead9d0badf216585aa90724b1da19d7c5e6/client/post-editor/controller.js#L283) somehow and I wasn't sure if there was a better place to override this string.

#### Testing instructions

1. Go to https://wordpress.com/page.
1. The text should now read "Select a site to start writing".
1. You should be able to select a site and create a new page as normal.

### CODE REVIEW

File: client/my-sites/sites/index.jsx [hunk 1]
```diff
@@ -78,6 +78,17 @@ export class Sites extends Component {
...
+		if ( path === 'page' ) {
+			return i18n.translate( 'Select a site to start writing', {
+				args: {
+					path: path,
+				},
+				components: {
+					strong: <strong />,
+				},
+			} );
+		}
```
Comment (Copons) [inline client/my-sites/sites/index.jsx]: We don't need the options object as the translation string doesn't have any variables to be replaced:

```js
return i18n.translate( 'Select a site to start writing' );
```

**Comment by Copons** [comment]
> Technical note: The code works, but I did notice that the Post wording seems to be coming from the Press This part of the post-editor controller somehow and I wasn't sure if there was a better place to override this string.

Would it work if we removed it from the post's Press This callback, and updated this PR to also include the post page?

E.g.
```js
if ( 'page' === path || 'post' === path ) {
  return i18n.translate( 'Select a site to start writing' );
}
```

**Comment by mmtr** [comment]
Just noted that we also display "Please select a site to open" when accessing Gutenlypso ( https://wordpress.com/block-editor). Should we handle this path as well?

image


> Would it work if we removed it from the post's Press This callback, and updated this PR to also include the post page?

Since [both paths use the same controller](https://github.com/Automattic/wp-calypso/blob/ee1f62933b55ba9de43c4a6267bfb835af478056/client/wordpress-com.js#L290-L296), we can also include a `controller.pressThis` callback in the `/page` route as [we do for the `/post` route](https://github.com/Automattic/wp-calypso/blob/ee1f62933b55ba9de43c4a6267bfb835af478056/client/post-editor/index.js#L16).

But this wouldn't work with Gutenberg, because `/block-editor` is using a different controller so I'd remove the logic from the post editor controller and move it to `Sites` as Jacopo suggested.

**Comment by codebykat** [comment]
Thanks @mmtr and @Copons! Updated to include the block-editor path and move everything into the site picker (plus clean up stray arguments and non-Yoda conditionals).

I am not sure how to test if the Press This part is still working as intended?

Also, I noticed the router also has an `edit` path -- presumably this only works, or is intended to work, when there's also a site and post ID. Right now visiting /edit by itself loads a blank screen.

**Comment by mmtr** [comment]
> Also, I noticed the router also has an edit path -- presumably this only works, or is intended to work, when there's also a site and post ID. Right now visiting /edit by itself loads a blank screen.

That's the route for the post custom types like Portfolio (https://wordpress.com/edit/jetpack-portfolio) or Testimonial (https://wordpress.com/edit/jetpack-testimonial). 

I think it makes sense to handle that path there, so I followed up on #29884.

**Pull request merged**